mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java

```diff
@@ -36,4 +37,6 @@
     Publisher<DeletedMessage> search(User user, Query query);
 
     Publisher<User> usersWithVault();
+
+    Task deleteExpiredMessagesTask();
```

Comment:
We definitely need to solve that Task/Publisher issue we have because it doesn't make sense to have two primitives to express the same thing (minus status update).

Reply:
Here we really care about the progress IMO